### PR TITLE
SAP Hana Integration Improvements

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR improves the SAP Hana integration and makes it enterprise-ready.

Fixes https://github.com/mindsdb/mindsdb_private/issues/639

## Type of change

- [X] 🐛 Bug fix (non-breaking change which fixes an issue)

## Verification Process

To ensure the changes are working as expected:

 - [X]   Test Location: `tests/unit/handlers/test_hana.py` for unit tests.
 - [X]   Verification Steps: Run the above test module.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/hana_handler/README.md [hunk 1]
```diff
@@ -1,74 +1,106 @@
...
-CREATE SCHEMA MINDSDB;
+Required connection parameters include the following:
 
-CREATE TABLE MINDSDB.TEST
-(
-    ID          INTEGER NOT NULL,
-    NAME        NVARCHAR(1),
-    DESCRIPTION NVARCHAR(1)
-);
+* `host`: The hostname, IP address, or URL of the SAP HANA database.
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/hana_handler/README.md:35]: In the code this is `address`, maybe we change it here?

File: mindsdb/integrations/handlers/hana_handler/README.md [hunk 2]
```diff
@@ -1,74 +1,106 @@
...
-* `port`: the port number of the SAP HANA instance
-* `user`: specifies the user name
-* `password`: specifies the password for the user
-* `schema`: sets the current schema, which is used for identifiers without a schema
+```sql
+CREATE DATABASE sap_hana_datasource
+WITH
+    ENGINE = 'hana',
+    PARAMETERS = {
+        "host": "123e4567-e89b-12d3-a456-426614174000.hana.trial-us10.hanacloud.ondemand.com",
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/hana_handler/README.md:25]: In the code this is `address`, maybe we change it here?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
One minor change

**Comment by MinuraPunchihewa** [comment]
> One minor change

@ZoranPandovski Do you think we should keep the connection parameter as `address` or change it `host`? The SDK refers to it as address, but we can make it host when accepting our arguments if necessary.

**Review by ZoranPandovski — APPROVED** [review]

**Comment by ZoranPandovski** [comment]
LEt's do the docs changes in another PR @MinuraPunchihewa

**Pull request merged**